cmd/frontend/internal/frontend/service.go

```diff
@@ -288,95 +293,79 @@ func (fes *FrontEndService) Monitor(ctx context.Context, errCh chan<- error) {
 				delay = 1 * time.Second
 			}
 
-			forced := logForced() // force status logs after config updates
 			protocolOut, err := fes.routingService.ShowProtocolSessions(ctx, lp, `NBR-*`)
 			if err != nil {
-				logger.Info("protocol output", "err", err, "out", strings.Split(protocolOut, "\n"))
-				//Note: if birdc is not yet running, no need to bail out
-				//linkCh <- "Failed to fetch protocol status"
-			} else if strings.Contains(protocolOut, bird.NoProtocolsLog) {
-				if !noConnectivity || init {
-					if refreshCancel != nil {
-						refreshCancel()
-					}
-					_ = denounceFrontend(fes.targetRegistryClient)
-					noConnectivity = true
-					connectivityMap = map[string]bool{}
-					logger.Info("protocol output", "out", protocolOut)
-					//linkCh <- "No protocols match"
+				logger.Error(err, "protocol output", "out", strings.Split(protocolOut, "\n"))
+				denounce = true
+				continue
+			}
+
+			if strings.Contains(protocolOut, bird.NoProtocolsLog) {
+				logger.Info("protocol output", "out", protocolOut)
+				denounce = true
+				continue
+			}
+
+			bfdOut, err := fes.routingService.ShowBfdSessions(ctx, lp, `NBR-BFD`)
+			if err != nil {
+				logger.Error(err, "BFD output", "out", strings.Split(bfdOut, "\n"))
+				denounce = true
+				continue
+			}
+
+			// determine status of gateways and connectivity
+			status := fes.parseStatusOutput(protocolOut, bfdOut)
+
+			// Gateway availibility notifications
+			if status.NoConnectivity() {
+				// although configured at least one IP family has no connectivity
+				if hasConnectivity {
+					denounce = true
+					// should this be moved to "if denounce" above?
```

Comment:
It could be moved there, yes. It reflects the status of external connectivity and is used by probes.